store/boltdbStore/business.go

```diff
@@ -18,52 +18,217 @@
+		return errors.New("Add Business missing some params")
+	}
+
+	dbOp := bs.handler
+
+	tNow := time.Now()
+	b.CreateTime = tNow
+	b.ModifyTime = tNow
+
+	lock := bs.lock
+	lock.Lock()
+	defer lock.Unlock()
+
+	// first : check is exist
+	old, _ := bs.lockfreeGetBusinessByID(b.ID)
+	if old != nil {
+		log.Errorf("[Store][database] already exist when do add business : %#v", b)
+		return errors.New("Add Business Duplicate")
+	}
+
+	// second : save business
+	if err := dbOp.SaveValue(DataTypeBusiness, b.ID, b); err != nil {
+		log.Errorf("[Store][database] add business err : %s", err.Error())
+		return store.Error(err)
+	}
+
 	return nil
 }
 
-// 删除一个业务集
-func (b *businessStore) DeleteBusiness(bid string) error {
-	//TODO
+// DeleteBusiness 删除一个业务集
+func (bs *businessStore) DeleteBusiness(bid string) error {
+	if bid == "" {
+		log.Errorf("[Store][database] delete business missing id")
+		return errors.New("Delete Business missing some params")
+	}
+
+	dbOp := bs.handler
+
+	lock := bs.lock
+	lock.Lock()
+	defer lock.Unlock()
+
+	if err := dbOp.DeleteValues(DataTypeBusiness, []string{bid}); err != nil {
+		log.Errorf("[Store][database] delete business err : %s", err.Error())
+		return store.Error(err)
+	}
+
 	return nil
 }
 
-// 更新业务集
-func (b *businessStore) UpdateBusiness(business *model.Business) error {
-	//TODO
+// UpdateBusiness 更新业务集
+func (bs *businessStore) UpdateBusiness(b *model.Business) error {
+	if b.ID == "" || b.Name == "" || b.Owner == "" {
+		log.Errorf("[Store][database] update business missing some params: %+v", b)
+		return errors.New("Update Business missing some params")
+	}
+
+	dbOp := bs.handler
+
+	b.ModifyTime = time.Now()
+
+	lock := bs.lock
+	lock.Lock()
+	defer lock.Unlock()
+
+	if err := dbOp.SaveValue(DataTypeBusiness, b.ID, b); err != nil {
+		log.Errorf("[Store][database] add business err : %s", err.Error())
+		return store.Error(err)
+	}
+
 	return nil
 }
 
-// 更新业务集token
-func (b *businessStore) UpdateBusinessToken(bid string, token string) error {
-	//TODO
-	return nil
+// UpdateBusinessToken 更新业务集token
+func (bs *businessStore) UpdateBusinessToken(bid string, token string) error {
+	if bid == "" || token == "" {
+		log.Errorf("[Store][business] update business token missing some params")
+		return errors.New("Update Business Token missing some params")
+	}
+
+	// first: get business by bid
+	b, err := bs.GetBusinessByID(bid)
+	if err != nil {
+		return store.Error(err)
+	}
+
+	b.Token = token
+	return bs.UpdateBusiness(b)
 }
 
-// 查询owner下业务集
-func (b *businessStore) ListBusiness(owner string) ([]*model.Business, error) {
-	//TODO
-	return nil, nil
+// ListBusiness 查询owner下业务集
+func (bs *businessStore) ListBusiness(owner string) ([]*model.Business, error) {
+	if owner == "" {
+		log.Errorf("[Store][business] list business missing owner")
+		return nil, errors.New("List Business Mising param owner")
+	}
+
+	result, err := bs.listAllBusiness()
```

Comment:
you shouldn't load all values here and filter it, you can use the LoadValuesByFilter to do the filtering

Reply:
already fix